### PR TITLE
ci(build): Add MacOS ARM CI job

### PR DESCRIPTION
**Pull request by warp-core**
**CI/CD/Testing**

## Summary
We don't actually test native ARM builds for MacOS, like, at all, until a release is made and ARM Mac users try the universal binary.
This PR adds a new CI job that builds and tests in a macos-14 GitHub hosted runner, which supposedly uses M1 hardware.

## Testing Done
It tests itself!

### CODE REVIEW

**Review by tibetiroka — APPROVED** [review]

**Review by petervdmeer — APPROVED** [review]
Looks like a good idea. Let's merge this (once the merge-conflict is resolved).

**Pull request merged**
